core/src/main/java/org/apache/carbondata/core/datastore/chunk/reader/dimension/v3/CompressedDimensionChunkFileBasedReaderV3.java

```diff
@@ -289,6 +289,10 @@ private DimensionColumnPage decodeDimensionLegacy(DimensionRawColumnChunk rawCol
     int[] rlePage;
     int[] invertedIndexes = new int[0];
     int[] invertedIndexesReverse = new int[0];
+    if (compressor == null) {
```

Comment:
Move this logic to CompressedDimChunkFileBasedPageLevelReaderV3.decodeColumnPage please refer CompressedDimensionChunkFileBasedReaderV3 .decodeColumnPage